src/java.base/share/classes/java/math/BigDecimal.java

```diff
@@ -1393,11 +1394,18 @@ static BigDecimal zeroValueOf(int scale) {
      * @since  1.5
      */
     public static BigDecimal valueOf(double val) {
-        // Reminder: a zero double returns '0.0', so we cannot fastpath
-        // to use the constant ZERO.  This might be important enough to
-        // justify a factory approach, a cache, or a few private
-        // constants, later.
-        return new BigDecimal(Double.toString(val));
+        if (!Double.isFinite(val)) {
+            throw new NumberFormatException("Infinite or NaN");
```

Comment:
Note, this changes the exception message. The new value is consistent with `new BigDecimal(double)`

Reply:
Yeas, we usually only rely on the exception type, not on the specific message, which is not part of the method spec.
While it is nice to keep the message unchanged over releases, it is sometimes hard to do so consistently.